test/models/concerns/key_pair_compute_resource_test.rb

```diff
@@ -0,0 +1,50 @@
+require 'test_helper'
+
+class DummyComputeResource < ComputeResource
+  include KeyPairComputeResource
+  include Mocha::API
+
+  def client
+    @client ||= mock('client')
+  end
+end
+
+class KeyPairComputeResourceTest < ActiveSupport::TestCase
+  test 'should remove the key pair on compute resource deletion' do
+    cr = DummyComputeResource.new
+    key_pair = FactoryGirl.build(:key_pair)
+    cr.key_pair = key_pair
+    mock_key_pairs = mock('mock_key_pairs')
+    fog_key_pair = mock('fog_key_pair')
+    cr.send(:client).expects(:key_pairs).returns(mock_key_pairs)
+    mock_key_pairs.expects(:get).with(key_pair.name).returns(fog_key_pair)
+    key_pair.expects(:destroy)
+    fog_key_pair.expects(:destroy)
+    assert cr.destroy!
+  end
```

Comment:
Nitpick: I think all of the mocks stubs should be created using `stubs` except of course `destroy` methods.